### PR TITLE
MM-37575: Handle nil value in override_username

### PR DESCRIPTION
**Pull request by agnivade**
Since props is a map[string]interface{}, an ", ok" check doesn't
really help because even if a key is not present, by default, a map
will always return the default value of the value, which would be a
nil interface.

So we can just check for the value of the map instead.

https://mattermost.atlassian.net/browse/MM-37575

```release-note
NONE
```

### CODE REVIEW

**Comment by agnivade** [comment]
Actually, this still has a bug. One moment.

**Comment by agnivade** [comment]
Ok all good now.

File: app/notification.go [hunk 1]
```diff
@@ -1068,8 +1068,10 @@ func (n *PostNotification) GetSenderName(userNameFormat string, overridesAllowed
 	}
 
 	if overridesAllowed && n.Channel.Type != model.ChannelTypeDirect {
-		if value, ok := n.Post.GetProps()["override_username"]; ok && n.Post.GetProp("from_webhook") == "true" {
-			return value.(string)
+		if value := n.Post.GetProps()["override_username"]; value != nil && n.Post.GetProp("from_webhook") == "true" {
+			if _, ok := value.(string); ok {
+				return value.(string)
```
Comment (isacikgoz) [inline app/notification.go:1073]: Would it make any difference to implement like this?
```suggestion
			if s, ok := value.(string); ok {
				return s
```

**Review by isacikgoz — APPROVED** [review]
🐛 -> 🔥

**Comment by agnivade** [comment]
https://github.com/mattermost/mattermost-server/pull/18297

**Pull request merged**